### ISSUE TITLE
[MU4 Issue] Invalid note is rendered on the TAB staff when the same note is added to 2+ voices

### ISSUE THREAD

**Issue by boryskuzmenko** [musescore/MuseScore#12176]
**Describe the bug**
TBC

**To Reproduce**
1. 
[Voices on tab test.mscz.zip](https://github.com/musescore/MuseScore/files/8990787/Voices.on.tab.test.mscz.zip)

**Expected behavior**

**Screenshots**
![Capture 2022-06-27_12-32-59_PM](https://user-images.githubusercontent.com/101250347/175914128-fc7f3ece-ca57-442b-b26f-1f8eb001122d.png)

**Platform information**
 - OS: [e.g. macOS, Windows, Linux]

**Additional context**

**Comment by HemantAntony** [comment]
Unless I'm missing something, isn't this correct already? Assuming the implementation of the last 3 notes in the 2nd measure is correct, the two voices play the same note in two different ways

But, when you look at the 1st note of the 2nd measure, there is no other way to play the E note other than the open note on the 6th string.

Similarly for the high note (17 on the 1st string). There is only one way to play that note

**Comment by abariska** [comment]
@HemantAntony I see your point but I believe that edge note shouldn't be marked as invalid when used by two or more voices. It makes no sense for me. Look at the more realistic example with these notes:
<img width="403" alt="CleanShot 2022-09-12 at 00 05 33" src="https://user-images.githubusercontent.com/101250347/189549529-728db0b8-b0d3-40b6-a772-4d5214a520da.png">
It could be timbre specific task to use this notes in both voices.
@bkunda may be you could add something as guitarist?

**Comment by bkunda** [comment]
@abariska I'm sorry but I'm not sure I understand what the issue is here that you're describing. 

Is it that these notes in red _shouldn't_ be marked as invalid?

**Comment by abariska** [comment]
Yep, I believe that when two voices use the same note it shouldn't be marked as invalid

**Comment by bkunda** [comment]
OK I think I see your point. 
And actually the question I would have here is whether the user is even likely to understand what the red highlight indicates. 
Certainly, it's not _invalid_ to have two voices written on a single note in guitar music, even for pitches that are technically only playable in one position (See below example, which is not uncommon in classical guitar music):
<img width="601" alt="Screenshot 2022-09-12 at 1 28 53 pm" src="https://user-images.githubusercontent.com/86290556/189643637-e8c4a0a1-fb6c-45f4-a6b7-0191414b4333.png">
Of course this kind of notation makes less sense in tablature, but I'm not sure that highlighting this adds much value to the user.

**Comment by tmr232** [comment]
As a user - I find this very confusing.
I just came here after having the red notes shown in a tab I imported from GP.
I could not understand what the red marking is (my guess was that it's warning about something unplayable - but they are were in a playable range). Took me some messing with the tab in GP to realize it's due to 2 notes in one position.

So for my - it mostly adds confusion.
Maybe could be better if mouse-over would indicate the issue, but as it is it is just confusing.

**Comment by abariska** [comment]
> As a user - I find this very confusing. I just came here after having the red notes shown in a tab I imported from GP. I could not understand what the red marking is (my guess was that it's warning about something unplayable - but they are were in a playable range). Took me some messing with the tab in GP to realize it's due to 2 notes in one position.
> 
> So for my - it mostly adds confusion. Maybe could be better if mouse-over would indicate the issue, but as it is it is just confusing.

Could you please provide the GP tab?

**Comment by tmr232** [comment]
Check bars 89, 91, 92

[Wicked - Defying Gravity(1).zip](https://github.com/musescore/MuseScore/files/13712460/Wicked.-.Defying.Gravity.1.zip)

**Comment by abariska** [comment]
@bkunda @oktophonie would be good to design and implement possibility to add two notes with the same pitch to the same string (and fret). As a result this issue will also be fixed https://github.com/musescore/MuseScore/issues/20968.

My proposition to implement these requirements:
- when creating two or more same notes with different voices on Standard staff, the notes should be placed on the same default string and fret on Tab linked staff, regardless the linked staff is already created or will be created later
- visually, on the Tab staff all the notes should look like one, like stack of notes
- when the fret (stack of frets to be more accurate) is selected, note with the last voice (2,3 or 4) should be selected. Switching between voices to be the same as for Standard staff (Cmd+click)
<img width="402" alt="abariska 2024-07-05 at 14 57 42" src="https://github.com/musescore/MuseScore/assets/101250347/d95618d7-0769-4ea1-a156-b9b8eb4d6e11">

<img width="405" alt="abariska 2024-07-05 at 15 16 26" src="https://github.com/musescore/MuseScore/assets/101250347/bad70063-d948-4655-9569-7e6bf11fcec4">

- regardless the number of used voices, each note of the stack can be moved to its particular string or they could be placed on one string
- all other functionality is the same for each note in the stack
